pkg/reconciler/replication/replication_reconcile_apiexports.go

```diff
@@ -0,0 +1,162 @@
+//  2. deletion of the object from the cache server
+//     happens when either of the following is true:
+//     - the original/local object was removed
+//     - the original/local object was not found in c.localApiExport
+//  3. modification of the object to match the original object
+//     happens when either of the following is true:
+//      - the original object's metadata doesn't match the cached one.
+//        as of today we compare only the annotations and the labels
+//      - the original object's spec doesn't match the cached one.
+//      - the original object's status doesn't match the cached one.
+func (c *controller) reconcileAPIExports(ctx context.Context, key string) error {
+	localApiExport, err := c.localApiExportLister.Get(key)
+	if errors.IsNotFound(err) {
+		clusterName, apiExportName := clusters.SplitClusterAwareKey(key)
+		cacheApiExport, err := c.retrieveApiExport(apiExportName, clusterName.String())
+		if err != nil {
+			return err
+		}
+		return c.handleApiExportDeletion(ctx, nil, cacheApiExport)
+	}
+	if err != nil {
+		return err
+	}
+	var cacheApiExport *apisv1alpha1.APIExport
+	cacheApiExport, err = c.retrieveApiExport(localApiExport.Name, logicalcluster.From(localApiExport).String())
+	if err != nil {
+		return err
+	}
+	if localApiExport.DeletionTimestamp != nil {
+		return c.handleApiExportDeletion(ctx, localApiExport, cacheApiExport)
+	}
+
+	if cacheApiExport == nil {
+		copyLocalApiExport := localApiExport.DeepCopy()
+		copyLocalApiExport.ResourceVersion = ""
+		_, err := c.cacheKcpClusterClient.Cluster(logicalcluster.From(copyLocalApiExport)).ApisV1alpha1().APIExports().Create(ctx, copyLocalApiExport, metav1.CreateOptions{})
+		return err
+	}
+
+	copyCacheApiExport := cacheApiExport.DeepCopy()
+	if copyCacheApiExport.Annotations != nil {
+		delete(copyCacheApiExport.Annotations, genericrequest.AnnotationKey)
+	}
+	objectMetaChanged := ensureObjectMeta(&copyCacheApiExport.ObjectMeta, localApiExport.ObjectMeta)
+	specChanged := ensureApiExportSpec(&copyCacheApiExport.Spec, localApiExport.Spec)
+	statusChanged := ensureApiExportStatus(&copyCacheApiExport.Status, localApiExport.Status)
+
+	if !specChanged && !objectMetaChanged && !statusChanged {
+		return nil
+	}
+
+	if specChanged || objectMetaChanged {
+		_, err = c.cacheKcpClusterClient.Cluster(logicalcluster.From(localApiExport)).ApisV1alpha1().APIExports().Update(ctx, copyCacheApiExport, metav1.UpdateOptions{})
+		return err
+	}
+
+	if !statusChanged {
+		return nil
+	}
+	_, err = c.cacheKcpClusterClient.Cluster(logicalcluster.From(localApiExport)).ApisV1alpha1().APIExports().UpdateStatus(ctx, copyCacheApiExport, metav1.UpdateOptions{})
+	return err
+}
+
+func (c *controller) retrieveApiExport(apiExportName, clusterName string) (*apisv1alpha1.APIExport, error) {
+	var cacheApiExport *apisv1alpha1.APIExport
+	cacheApiExports, err := c.cacheApiExportsIndexer.ByIndex(ByShardAndLogicalCluster, toShardAwareKey(c.shardName, clusterName))
```

Comment:
ah ack you do share the formatter, but if you export one, let's export the other